### PR TITLE
Generic Provide + Notifications

### PR DESCRIPTION
**Pull request by BKSteve**
Fix Generic Provider `episode_object` error
#8688

## Summary by CodeRabbit

- **New Features**
	- Added `DISCORD_WEBHOOK` variable to settings for enhanced Discord integration.
	- Updated URL for "opensubtitles" source to use HTTPS for improved security.
- **Bug Fixes**
	- Corrected the logic in the `get_coming_episodes` function to ensure it always returns results.
- **Refactor**
	- Simplified Discord notification logic by removing the `tts` parameter and related settings.
	- Improved error logging in Discord notification sending.
- **Chores**
	- Removed the "Discord TTS" checkbox from the GUI to streamline notification settings.
- **Style**
	- Adjusted search results retrieval logic in `GenericProvider.py` for clarity and efficiency.

### CODE REVIEW

**Comment by BKSteve** [comment]
Remove Discord TTS setting and code as it doesn't TTS and message rather than speech is sufficient notification.
Otherwise requires much bigger rewrite.

**Comment by BKSteve** [comment]
Add in change of `http://www.opensubtitles.org` to `https://www.opensubtitles.com`

File: sickchill/views/config/notifications.py [hunk 1]
```diff
@@ -368,7 +367,6 @@ def saveNotifications(
         settings.DISCORD_WEBHOOK = discord_webhook
         settings.DISCORD_NAME = discord_name
         settings.DISCORD_AVATAR_URL = discord_avatar_url
-        settings.DISCORD_TTS = discord_tts
```
Comment (miigotu) [inline sickchill/views/config/notifications.py:371]: This is the only line needed changed for discord_tts, to `settings.DISCORD_TTS = config.checkbox_to_value(discord_tts)`

Tts is for blind people, and we have sight-impaired users. They just don't talk much.

**Comment by BKSteve** [comment]
`Discord TTS big change` fixes the TTS setting and realigns it and improves selection to enable the whole message or just the bot name to be sent.
The Bot name can be anything you want as actual bot is based on web hook. It just can't be blank so it gets forced to SickChill if blank.

With this testing have added in `index.py` try around `await self.finish(results)` as was throwing an exception with all shows regarding `prowl_notify_list` which core.js says is an `Unresolved variable` on 1228.
Need some assistance with this js please.

File: sickchill/gui/slick/js/core.js [hunk 2]
```diff
@@ -716,11 +716,11 @@ const SICKCHILL = {
 
                 discord.name = $('#discord_name').val();
                 discord.avatar = $('#discord_avatar_url').val();
-                discord.tts = $('#discord_tts').val();
+                discord.tts = $('#discord_tts').is(':checked') ? 1 : 0;
```
Comment (miigotu) [inline sickchill/gui/slick/js/core.js]: You don't need the ternary here, you can just do:
```javascript
$('#discord_tts').is(':checked');
```

`checkbox_to_value` will handle the correct conversion.

Comment (BKSteve) [reply sickchill/gui/slick/js/core.js]: Saw this used elsewhere in core.js and as I couldn't get tts to work with the returned value and store this aligned with the same `true` or `false` values.  
I therefore went with 1 or 0 like many other settings and just simple int when passed from mako, js, python and
`settings.DISCORD_TTS = check_setting_bool(settings.CFG, "Discord", "discord_tts")` as stored are all aligned.

**Pull request merged**